### ISSUE TITLE
Login does not work with disableLocalStrategy: true

### ISSUE THREAD

**Issue by JesperWe** [WilsonLe/payload-oauth2#18]
Thanks a lot for making this plugin available!
I am using Payload v3 Beta 118.
After some adaptions of your Google example I have successfully set it up using a custom identity provider, Zitadel in this case.
It works well as long as I do not disable Payloads built-in authentication. 
But if I add 
```
      auth: {
        disableLocalStrategy: true,
      },
```
to the users collection config in order to hide Payloads own login fields, the OAuth2 login no longer works. Instead of ending up at the /admin url after a successful login I now end up back on /admin/login and so I am stuck in a loop.

This is my current config:
```js
export default buildConfig({
  editor: lexicalEditor(),
  collections: [
    {
      slug: 'users',
      auth: {
        disableLocalStrategy: true,
      },
      admin: {
        useAsTitle: 'email',
        defaultColumns: ["email", "name"]
      },
      access: {
        delete: () => false,
        update: () => false,
      },
      fields: [
        { name: "email", type: "email" },
        { name: "name", type: "text" }
      ],
    },
    {
      slug: 'pages',
      admin: {
        useAsTitle: 'title',
      },
      fields: [
        {
          name: 'title',
          type: 'text',
        },
        {
          name: 'content',
          type: 'richText',
        },
      ],
    },
    {
      slug: 'media',
      upload: true,
      fields: [
        {
          name: 'text',
          type: 'text',
        },
      ],
    },
  ],
  secret: process.env.PAYLOAD_SECRET || '',
  typescript: {
    outputFile: path.resolve(dirname, 'payload-types.ts'),
  },
  db: postgresAdapter({
    pool: {
      connectionString: process.env.DATABASE_URI,
      ssl: {rejectUnauthorized: false},
    },
  }),

  i18n: {
    supportedLanguages: { en, sv },
  },

  admin: {
    importMap: { baseDir: path.resolve(dirname) },
    components: {
      beforeLogin: ["/src/components/OAuthLoginButton"],
    },
    user: "users",
  },

  plugins: [
    OAuth2Plugin({
      enabled: true,
      strategyName: "Zitadel",
      serverURL: process.env.NEXT_PUBLIC_URL || "http://localhost:3000",
      authCollection: "users", // assuming you already have a users collection with auth enabled
      clientId: process.env.OIDC_CLIENT_ID || "",
      clientSecret: process.env.OIDC_CLIENT_SECRET || "",
      tokenEndpoint: process.env.OIDC_TOKEN_URI ?? '',
      scopes: [
        "openid",
        "profile",
        "email",
        "offline_access",
        "urn:zitadel:iam:user:metadata",
      ],
      providerAuthorizationUrl: process.env.OIDC_AUTH_URI ?? "",
      getUserInfo: async (accessToken: string) => {
        const response = await fetch(
          process.env.OIDC_USERINFO_URI ?? "",
          { headers: { Authorization: `Bearer ${accessToken}` } },
        );
        const user = await response.json();
        return { email: user.email, sub: user.sub, name: user.name };
      },
      successRedirect: () => "/admin",
      failureRedirect: () => "/admin/login",
    }),
  ],
  
  sharp,
})
```

**Comment by WilsonLe** [comment]
can you add change modify your config at `failureRedirect` to be

```ts
failureRedirect: (error) => {
    console.error(error);
    return "/admin/login";
}
```

**Comment by WilsonLe** [comment]
I'll modify the example to log errors by default

**Comment by JesperWe** [comment]
Please re-open this @WilsonLe, the issue is not solved.

I have created a new project from scratch with up to date payload (3.4.0) and this plugin updated.
Oauth login works fine until I `disableLocalStartegy`, then I just get redirected back to the login page after the Oauth sequence.

**Comment by JesperWe** [comment]
In the screenshot below you can see a sequence of requests, starts by going to` /admin` which redirects to` /login`, Oauth is triggered from there and goes of to the external Service Provider system, then comes back to `/callback`, after which it again goes to `/admin` and `/login`.

This all works fine when the LocalStrategy is NOT disabled. There are no errors printed on the browser console or on the pnpm console.

<img width="824" alt="image" src="https://github.com/user-attachments/assets/5b1aa374-4699-4bbe-a0ff-a95daadb5837">

**Comment by WilsonLe** [comment]
Interesting, let me try to reproduce this.

**Comment by WilsonLe** [comment]
Weird, I even tried using Zitadel and it seems to work, even with `disableLocalStrategy: true`. You can see the test I setup. Can you do a `pnpm reinstall` to ensure everything is up-to-date?

**Comment by JesperWe** [comment]
There was a couple of fixes merged to external auth since 3.4.0 so I tried 3.7.0 but same result.
The annoying thing is that everything works as long as I have `disableLocalStrategy` undefined.

I'll try to get you some more context:
Looking at my logs, the redirect from Zitadel to `localhost:3000/api/users/oauth/callback` seems fine, it responds 302 with a `set-cookie` header for `payload-token` and a redirect to `/admin`.

The following `/admin` looks fine too, it has the `payload-token` cookie, but:
- With `disableLocalStrategy` undefined the response is 200 with the correct page. 
- With `disableLocalStrategy: true` I get a 307 redirect to `location: /admin/login`.

I added some debug to the plugin config, with this code
```typescript
    OAuth2Plugin({
      enabled: true,
      serverURL: process.env.NEXT_PUBLIC_URL || 'http://localhost:3000',
      authCollection: 'users', // assuming you already have a users collection with auth enabled
      clientId: process.env.CLIENT_ID || '',
      clientSecret: process.env.CLIENT_SECRET || '',
      tokenEndpoint: process.env.TOKEN_URI || '',
      scopes: [
        'openid',
        'profile',
        'email',
        'urn:iam:org:project:roles',
      ],
      prompt: 'select_account',
      providerAuthorizationUrl: process.env.AUTH_URI ?? '',
      getUserInfo: async (accessToken: string) => {
        console.log("______________getUserInfo________________")
        const response = await fetch(
          process.env.USERINFO_URI || '',
          { headers: { Authorization: `Bearer ${accessToken}` } },
        )
        const user = await response.json()
        const tokenRoles = user?.['urn:zitadel:iam:org:project:roles'] ?? {}
        const roleNames = Object.keys(tokenRoles)
        console.log({ roleNames })
        return { email: user.email, sub: user.sub, roles: roleNames }
      },
      successRedirect: (req) => {
        console.log("______________successRedirect________________")
        return '/admin'
      },
      failureRedirect: (req, error) => {
        console.log("______________failureRedirect________________")
        console.error(error)
        return '/login'
      },
      strategyName: 'zitadel',
    }),
```

I get this log:

```
 GET /api/users/oauth/authorize 302 in 7ms
______________getUserInfo________________
{ roleNames: [ 'admin', 'webmaster' ] }
______________successRedirect________________
 GET /api/users/oauth/callback?code=tghtbAb4IIz3qfmUOuS1t3WjYs_55_wHwy8_v6AbvPrlcw 302 in 1957ms
 GET /admin 307 in 382ms
 GET /admin/login 200 in 32ms
 GET /api/users/me 200 in 6ms
```

Any other things ideas for what I could check?

**Comment by JesperWe** [comment]
GAH just realized that the only problem is I was missing the `useEmailAsIdentity` setting. 

...wondering if there would be a more gratious way to handle this. Flag an error somewhere or something...
